src/unpriv-cfi.adoc

```diff
@@ -762,11 +762,11 @@ back_cfi_not_active:
 ], config:{lanes: 1, hspace:1024}}
 ....
 
-For RV32, `SSAMOSWAP.W` atomically loads a 32-bit data value from address of a
-shadow stack location in `rs1`, puts the loaded value into register `rd`, and
-stores the 32-bit value held in `rs2` to the original address in `rs1`.
-`SSAMOSWAP.D` (RV64 only) is similar to `SSAMOSWAP.W` but operates on 64-bit
-data values.
+For RV32, `SSAMOSWAP.W`, if granted access according to page protections rules,
```

Comment:
I am not sure the addition of "if granted access according to page protection rules" is required. All memory accesses are subject to page protection rules, PMP/PMA rules, Svukte rule, and other conditions such as those that cause hardware error exception (attempts to consume poison). Since this principle that memory access occurs only after everything that prevents a memory access is checked, this addition may not be required.

Reply:
You are right; it might not be the right term. I was trying to highlight that the pseudocode below only provides information for some exceptions. But it is not exhaustive, and these SSAMO operations might fail prior to the X(rd) = mem[X(rs1] ...

Maybe the best solution is to split the "illegal/virtual exception part" and the swap /sign-swap pseudocode what do you think?

Reply:
Sorry, I somehow dropped this thread. My suggestion is to just drop this addition since its stating the obvious which i.e. memory accesses need to be granted access to be allowed.